internal/backends/postgresql/helpers.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package postgresql
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"golang.org/x/exp/maps"
+
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+)
+
+// unmarshalExplain unmarshalls the plan from EXPLAIN postgreSQL command.
+// EXPLAIN result is not sjson, so it cannot be unmarshalled by sjson.Unmarshal.
+func unmarshalExplain(b []byte) (*types.Document, error) {
```

Comment:
Can we implement this similarly to rest of the code in backend by implementing `Marshal`  `Unmarshal` interface?

Reply:
I think I don't understand. Can you show an example please?

Reply:
Sorry my explanation wasn't clear :disappointed:

There is only one row, we don't need iterator. See this diff if it makes sense :pray: 
 https://github.com/FerretDB/FerretDB/pull/3465#discussion_r1343323383